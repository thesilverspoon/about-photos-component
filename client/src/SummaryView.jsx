import React from 'react';
import ajax from './ajax.js';
import About from './About.jsx';
import SideBar from './SideBar.jsx';
import Banner from './Banner.jsx';

class SummaryView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      id: props.id.id,
      name: props.id.name,
      description: props.id.about.description,
      hours:  props.id.about.hours,
      phone: props.id.about.phone,
      price: props.id.about.price,
      style: props.id.about.style,
      banner: props.id.banner,
      photo: props.id.photo,
    }
  }

  // componentDidMount() {
  //   this.fetch();
  // }

  // fetch() {
  //   ajax.get(this.state.id, (err, data) => {
  //     if (err) {
  //       throw err;
  //     } else {
  //       this.setState({
  //         name: data[0].name,
  //         description: data[0].about.description,
  //         hours: data[0].about.hours,
  //         phone: data[0].about.phone,
  //         price: data[0].about.price,
  //         style: data[0].about.style,
  //         banner: data[0].banner,
  //         photo: data[0].photo
  //       })
  //     }
  //   });
  // }

  render() {
    return (
      <div className="grid">
        <div className="banner"><Banner banner={this.state.banner} /></div>
        <div className="main">
          <div className="about"><About description={this.state.description} name={this.state.name} photo={this.state.photo} /></div>
        </div>
        <div className="sidebar">
          <div><SideBar hours={this.state.hours} phone={this.state.phone} price={this.state.price} style={this.state.style} /></div>
        </div>
      </div>
    );
  }
}

export default SummaryView;
