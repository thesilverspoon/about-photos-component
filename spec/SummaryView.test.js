import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SummaryView from '../client/src/SummaryView.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('SummaryView Component', () => {
  test('should render correctly', () => {
    expect(Enzyme.shallow(<SummaryView />)).toMatchSnapshot();
  });
});
