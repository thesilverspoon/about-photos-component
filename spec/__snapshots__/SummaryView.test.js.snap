// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryView Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SummaryView />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="banner"
        >
          <Banner
            banner={Array []}
          />
        </div>,
        <div
          className="main"
        >
          <div
            className="about"
          >
            <About
              description=""
              name=""
              photo={Array []}
            />
          </div>
        </div>,
        <div
          className="sidebar"
        >
          <div>
            <SideBar
              hours=""
              phone=""
              price=""
              style=""
            />
          </div>
        </div>,
      ],
      "className": "grid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Banner
            banner={Array []}
          />,
          "className": "banner",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "banner": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="about"
          >
            <About
              description=""
              name=""
              photo={Array []}
            />
          </div>,
          "className": "main",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <About
              description=""
              name=""
              photo={Array []}
            />,
            "className": "about",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "description": "",
              "name": "",
              "photo": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <SideBar
              hours=""
              phone=""
              price=""
              style=""
            />
          </div>,
          "className": "sidebar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <SideBar
              hours=""
              phone=""
              price=""
              style=""
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "hours": "",
              "phone": "",
              "price": "",
              "style": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="banner"
          >
            <Banner
              banner={Array []}
            />
          </div>,
          <div
            className="main"
          >
            <div
              className="about"
            >
              <About
                description=""
                name=""
                photo={Array []}
              />
            </div>
          </div>,
          <div
            className="sidebar"
          >
            <div>
              <SideBar
                hours=""
                phone=""
                price=""
                style=""
              />
            </div>
          </div>,
        ],
        "className": "grid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Banner
              banner={Array []}
            />,
            "className": "banner",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "banner": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="about"
            >
              <About
                description=""
                name=""
                photo={Array []}
              />
            </div>,
            "className": "main",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <About
                description=""
                name=""
                photo={Array []}
              />,
              "className": "about",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "description": "",
                "name": "",
                "photo": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <SideBar
                hours=""
                phone=""
                price=""
                style=""
              />
            </div>,
            "className": "sidebar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <SideBar
                hours=""
                phone=""
                price=""
                style=""
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "hours": "",
                "phone": "",
                "price": "",
                "style": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
